OrbitVulkanLayerClientGgp/OrbitCaptureClientLayer.cpp

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <assert.h>
+#include <string.h>
+
+#include "OrbitBase/Logging.h"
+#include "OrbitVulkanLayerClientGgp/DispatchTable.h"
+#include "absl/base/casts.h"
+#include "absl/container/flat_hash_map.h"
+#include "absl/container/flat_hash_set.h"
+#include "absl/synchronization/mutex.h"
+#include "vulkan/vk_layer.h"
+#include "vulkan/vk_layer_dispatch_table.h"
+#include "vulkan/vulkan.h"
+
+#undef VK_LAYER_EXPORT
+#if defined(WIN32)
+#define VK_LAYER_EXPORT extern "C" __declspec(dllexport)
+#else
+#define VK_LAYER_EXPORT extern "C"
+#endif
+
+// Layer information
+static constexpr char const* kLayerName = "VK_LAYER_ORBIT_CAPTURE_CLIENT";
+static constexpr char const* kLayerDescription =
+    "Layer that contains Orbit Client implementation to run captures";
+static constexpr uint32_t const kLayerImplementationVersion = 1;
+static constexpr uint32_t const kLayerSpecVersion = VK_API_VERSION_1_1;
+
+// Vulkan layer
+absl::Mutex layer_mutex;
+DispatchTable dispatch_table;
+
+// --------------------------------------------------------------------------------
+// Layer init and shutdown
+// --------------------------------------------------------------------------------
+
+VK_LAYER_EXPORT VkResult VKAPI_CALL
```

Comment:
Oh, really? I was not aware of this; do you know why do they use it in the example tutorial?

Reply:
Probably to have uniform code. The effect of the macro is really, just to have the symbol names as C style plain. So it would be really just be `OrbitCaptureClientCreateInstance`. This is required by all the functions, the loader needs to call directly, which are only the Get*ProcAddr once (as all others can be retrieved by those two).
So there is really no need in having this on all vulkan functions

Reply:
Oh I get it, I have updated the code, thanks!